recipes/cpc2/baseline/predict.py

```diff
@@ -0,0 +1,107 @@
+"""Make intelligibility predictions from HASPI scores."""
+import json
+import logging
+from pathlib import Path
+from typing import Optional
+
+import hydra
+import numpy as np
+import pandas as pd
+from omegaconf import DictConfig
+from scipy.optimize import curve_fit
+
+from clarity.utils.file_io import read_jsonl
+
+log = logging.getLogger(__name__)
+
+
+class LogisticModel:
+    """Class to represent a logistic mapping.
+
+    Fits a logistic mapping from input values x to output values y.
+    """
+
+    params: Optional[np.ndarray] = None  # The model params
+
+    def _logistic_mapping(self, x, x_0, k):
+        """
+        Logistic function
+            x_0 - x value of the logistic's midpoint
+            k - the logistic growth rate or steepness of the curve
+        """
+        return 100.0 / (1 + np.exp(-k * (x - x_0)))
+
+    def fit(self, x, y):
+        """Fit a mapping from x values to y values."""
+        initial_guess = [0.5, 1.0]  # Initial guess for parameter values
+        self.params, *_pcov = curve_fit(self._logistic_mapping, x, y, initial_guess)
+
+    def predict(self, x):
+        """Predict y values given x."""
+        # Note, fit() must be called before predictions can be made
+        assert self.params is not None
+        return self._logistic_mapping(x, self.params[0], self.params[1])
+
+
+def make_disjoint_train_set(
+    full_df: pd.DataFrame, test_df: pd.DataFrame
+) -> pd.DataFrame:
+    """Make a disjoint train set for given test samples."""
+    # make sure that the train and test sets are disjoint
+    # i.e. no signals, systems or listeners are shared
+    train_df = full_df[~full_df.signal.isin(test_df.signal)]
+    train_df = train_df[~train_df.system.isin(test_df.system)]
+    train_df = train_df[~train_df.listener.isin(test_df.listener)]
+    assert not set(train_df.signal).intersection(set(test_df.signal))
+    return train_df
+
+
+# pylint: disable = no-value-for-parameter
+@hydra.main(config_path=".", config_name="config")
+def predict(cfg: DictConfig):
+    """Predict intelligibility from HASPI scores."""
+
+    # Load the intelligibility dataset records
+    dataset_filename = Path(cfg.path.metadata_dir) / f"{cfg.dataset}.json"
+    with open(dataset_filename, "r", encoding="utf-8") as fp:
```

Comment:
As above, very minor but can use the fact that `dataset_filename` is of type `Path`...

```suggestion
    with dataset_filename.open("r", encoding="utf-8") as fp:
```

Reply:
Dealing with this one now so that we have some best practice to follow. We can then work on improving recipes later.